src/librustdoc/clean/mod.rs

```diff
@@ -1967,18 +1967,22 @@ impl Clean<Span> for rustc_span::Span {
             return Span::empty();
         }
 
+        // Check if the span is the result of a macro expansion,
+        // so we can point to the invocation instead of the definition. (See rust-lang/rust#39726)
+        let span = if self.from_expansion() { self.source_callsite() } else { *self };
```

Comment:
The `from_expansion` check isn't necessary I think - if it's not from an expansion, `source_callsite` will return the original span: https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_span/lib.rs.html#444-447
```suggestion
        let span = self.source_callsite();
```